src/nodes.coffee

```diff
@@ -497,6 +523,103 @@ exports.Block = class Block extends Base
         fragments.push @makeCode "\n"
     fragments.concat post
 
+  compileComments: (fragments) ->
+    for fragment, fragmentIndex in fragments
+      # Insert comments into the output at the next or previous newline.
+      # If there are no newlines at which to place comments, create them.
+      if fragment.precedingComments
+        # Determine the indentation level of the fragment that we are about
+        # to insert comments before, and use that indentation level for our
+        # inserted comments. Because some fragments begin with a space, e.g.
+        # ` + 1`, we look for either a tab character or at least two spaces
+        # to detect indentation. If someone is actually using one-space
+        # indentation, well, they’ll get unindented line comments.
+        fragmentIndent = ''
+        for pastFragment in fragments[0...(fragmentIndex + 1)] by -1
+          indent = /^\t+| {2,}/.exec pastFragment.code
```

Comment:
Sometimes an indent appears not at the beginning of a token’s `code` property, such as after a comma for an array: `,\n\s\s` (using `\s` because GitHub swallows the spaces). 

`/^ {2,}/m` catches everything as far as I can tell.